app/cmd/rpc/mesh.go

```diff
@@ -589,67 +677,122 @@ func reloadChains(chainsPath string) {
 }
 
 // reloadServicers - reload servicersName file
-func reloadServicers(servicersPath string) {
-	if !fileExist(servicersPath) {
-		log2.Println(fmt.Sprintf("servicers file not found at %s; ignoring reload", servicersPath))
-		return
-	}
-
-	newServicers := getServicersFromFile()
-	newServicersMap := map[string]bool{}
+func reloadServicers() {
+	nodes, servicers := loadServicersFromFile()
 
-	for i := range newServicers {
-		ns := &newServicers[i]
+	currentNodes := mapset.NewSet[string]()
+	currentServicers := mapset.NewSet[string]()
+	reloadedNodes := mapset.NewSet[string]()
+	reloadedServicers := mapset.NewSet[string]()
 
-		pk, err := crypto.NewPrivateKey(ns.PrivateKey)
-		if err != nil {
-			log2.Fatal(fmt.Errorf("error parsing private key at index=%d of the file %s", i, servicersPath))
-		}
+	nodesMap.Range(func(key, _ any) bool {
+		currentNodes.Add(key.(string))
+		return true
+	})
+	nodes.Range(func(key, _ any) bool {
+		reloadedNodes.Add(key.(string))
+		return true
+	})
+	servicerMap.Range(func(key, _ any) bool {
+		currentServicers.Add(key.(string))
+		return true
+	})
+	servicers.Range(func(key, _ any) bool {
+		reloadedServicers.Add(key.(string))
+		return true
+	})
 
-		address, err := sdk.AddressFromHex(pk.PubKey().Address().String())
-		if err != nil {
-			log2.Fatal(fmt.Errorf("error getting address from private key at index=%d of the file %s", i, servicersPath))
-		}
+	newNodes := reloadedNodes.Difference(currentNodes)
+	newServicers := reloadedServicers.Difference(currentServicers)
 
-		addressStr := address.String()
+	removedNodes := currentNodes.Difference(reloadedNodes)
+	removedServicers := currentServicers.Difference(reloadedServicers)
 
-		newServicersMap[addressStr] = true
-	}
+	orphanServicers := sync.Map{}
 
-	// looking for removed pk if any
-	mutex.Lock()
-	removedAddresses := make([]string, 0)
-	for _, address := range servicerList {
-		if _, ok := newServicersMap[address]; ok {
-			// still there
-			continue
+	removedNodes.Each(func(s string) bool {
+		if v, ok := nodesMap.LoadAndDelete(s); ok {
+			n := v.(*fullNode)
+			for i := range n.Servicers {
+				address := n.Servicers[i].Address.String()
+				if v, ok := servicerMap.LoadAndDelete(address); ok {
+					// so we can use same servicer if basically was a move from node A to node B
+					orphanServicers.Store(n.Servicers[i].Address.String(), v)
+				}
+			}
+			stopNode(v.(*fullNode))
 		}
+		return true
+	})
+	removedServicers.Each(func(address string) bool {
+		if v, ok := servicerMap.LoadAndDelete(address); ok {
+			s := v.(*servicer)
+			for i := range s.Node.Servicers {
```

Comment:
what does this do?

Reply:
remove servicers deleted (they were on the mesh before reloading the file and with the new file they are gone) from those nodes where they were because an option is that user remove a key from a fullNode but not the entire fullNode, or even move to a new fullNode so that it will remove from that fullNode to an "orphan" map so if there is another fullNode that is new, it will check first on orphan to grab the previous servicer instance, avoiding the need to reload the service app session cache again from remote fullNode on the first relay or throw errors on success relay notifications.

Reply:
resolved with the new xsync and rework after it